### PR TITLE
Jg/alp 130 link robot node with lidar subscribing node

### PR DESCRIPTION
**Pull request by juliagranato**
edit: bugs are gone and it works now!

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,11 @@
 cmake_minimum_required(VERSION 3.5)
 project(slam_publisher)
 
+# Add the FastSLAM directory
 add_subdirectory(FastSLAM)
+add_library(LidarSubscriberLib src/lidar_subscriber.cpp)
```
Comment (WillWu88) [inline CMakeLists.txt:6]: I don't know if it's best practice to add a single source file that's meant to be run to a separate library. For now we can keep it, but this should be refactored in the future.

File: include/lidar_subscriber.h
Comment (WillWu88) [inline include/lidar_subscriber.h:1]: Why are there two files that does the same thing with slightly different names?

File: src/build/.built_by
Comment (WillWu88) [inline src/build/.built_by:1]: Please delete this and the "install" and "build" folder. Looks like you invoked colcon in the wrong directory.

File: src/lidar-subscriber.h
Comment (WillWu88) [inline src/lidar-subscriber.h:1]: Why is there another header file here that basically does the same thing as the last two?

File: src/lidar_subscriber.cpp [hunk 2]
```diff
@@ -96,46 +109,11 @@ std::queue<Observation2D> LidarSubscriber::laserscan_to_landmarks(const sensor_m
     merge_landmarks(landmarks.front(), landmarks.back(), counts.front(), counts.back());
     landmarks.pop();
   }
-  RCLCPP_DEBUG(this->get_logger(), "%u %f %f\n", landmarks.size(), landmarks.front().range_m, landmarks.front().bearing_rad);
-  #ifdef VISUALIZE_LANDMARKS
+
+  std::cout<<landmarks.size()<<"\n";
```
Comment (WillWu88) [inline src/lidar_subscriber.cpp]: Please replace this with the ROS logging framework

File: src/l
Comment (WillWu88) [inline src/l:1]: Is this made and included by mistake?

**Review by WillWu88 — CHANGES_REQUESTED** [review]
Please clean the source directory and your changes... way too much junk artifacts and files included here. Thanks

File: src/l
Comment (juliagranato) [reply src/l:1]: I believe so. It has been deleted

File: include/lidar_subscriber.h
Comment (juliagranato) [reply include/lidar_subscriber.h:1]: Not sure... there is only one now

File: CMakeLists.txt (hunk 1 above)
Comment (juliagranato) [reply CMakeLists.txt:6]: I totally agree. I was really struggling to properly link the header and source files and tried a bunch of things. For some reason that was the only thing that worked so I left it

File: src/lidar_subscriber.cpp (hunk 2 above)
Comment (juliagranato) [reply src/lidar_subscriber.cpp]: Done!

**Review by eslaght — APPROVED** [review]

**Review by eslaght — APPROVED** [review]

**Pull request merged**